src/meta/processors/admin/BalancePlan.h

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef META_ADMIN_BALANCEPLAN_H_
+#define META_ADMIN_BALANCEPLAN_H_
+
+#include <gtest/gtest_prod.h>
+#include "kvstore/KVStore.h"
+#include "meta/processors/admin/BalanceTask.h"
+
+namespace nebula {
+namespace meta {
+
+class BalancePlan {
+    friend class Balancer;
+    FRIEND_TEST(BalanceTest, BalancePlanTest);
+    FRIEND_TEST(BalanceTest, NormalTest);
+    FRIEND_TEST(BalanceTest, RecoveryTest);
+    FRIEND_TEST(BalanceTest, DispatchTasksTest);
+
+public:
+    enum class Status : uint8_t {
+        NOT_START          = 0x01,
+        IN_PROGRESS        = 0x02,
+        SUCCEEDED          = 0x03,
+        /**
+         * TODO(heng): Currently, after the plan failed, we will try to resume it
+         * when running "balance" again. But in many cases, the plan will be failed
+         * forever, it this cases, we should rollback the plan.
+         * */
+        FAILED             = 0x04,
+    };
+
+    BalancePlan(BalanceID id, kvstore::KVStore* kv, AdminClient* client)
+        : id_(id)
+        , kv_(kv)
+        , client_(client) {}
+
+    void addTask(BalanceTask task) {
```

Comment:
why parameter is not BalanceTask &&task?

Reply:
Because I want to accept const reference and right reference both.